hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -581,7 +582,11 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
     LOG.info("Started block pruning ...");
     boolean isDistributedPruningEnabled = CarbonProperties.getInstance()
         .isDistributedPruningEnabled(carbonTable.getDatabaseName(), carbonTable.getTableName());
-    if (isDistributedPruningEnabled) {
+    if (filter.isEmpty() && filter.getExpression() != null &&
```

Comment:
can you do this once you get the filter itself ? I mean after line 570

Reply:
Modified